### PR TITLE
Task scheduler: add replication notify for new task to launch ASAP

### PR DESCRIPTION
**Pull request by MatMaul**
It also:
- remove `first_launch` parameter for the launched task, it is not that useful and can be emulated by storing a boolean in `result`.
- remove `task_is_running` since it needs to be run on the good worker.
- change `delete_task` to forbid deletion of active tasks (instead of running ones), so this function can be called from anywhere and not only on the good worker.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/replication/tcp/handler.py [hunk 1]
```diff
@@ -663,6 +669,15 @@ def on_LOCK_RELEASED(
...
         )
 
+    async def on_NEW_ACTIVE_TASK(
+        self, conn: IReplicationConnection, cmd: NewActiveTaskCommand
+    ) -> None:
+        """Called when get a new NEW_ACTIVE_TASK command."""
+        if self._task_scheduler:
+            task = await self._task_scheduler.get_task(cmd.data)
+            if task and task.status == TaskStatus.ACTIVE:
+                await self._task_scheduler._launch_task(task)
```
Comment (clokep) [inline synapse/replication/tcp/handler.py:679]: If this is command is only sent for new *active* task, why would its status not be active?

**Review by clokep — COMMENTED** [review]
Seems reasonable overall, I have one question though.

File: synapse/replication/tcp/handler.py (hunk 1 above)
Comment (MatMaul) [reply synapse/replication/tcp/handler.py:679]: It's just a sanity check, it shouldn't really happen indeed. I'll nuke it.

**Review by clokep — APPROVED** [review]
Looks reasonable. There's a lint error FYI.

**Pull request merged**